ksp/src/main/kotlin/net/skyscanner/backpack/ksp/BackpackSymbolProcessor.kt

```diff
@@ -0,0 +1,75 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.ksp
+
+import androidx.room.compiler.processing.ExperimentalProcessingApi
+import androidx.room.compiler.processing.XProcessingEnv
+import com.google.devtools.ksp.processing.Resolver
+import com.google.devtools.ksp.processing.SymbolProcessor
+import com.google.devtools.ksp.processing.SymbolProcessorEnvironment
+import com.google.devtools.ksp.processing.SymbolProcessorProvider
+import com.google.devtools.ksp.symbol.KSAnnotated
+import com.google.devtools.ksp.validate
+import net.skyscanner.backpack.ksp.visitor.ComponentsVisitor
+import net.skyscanner.backpack.ksp.visitor.ComposeStoriesVisitor
+import net.skyscanner.backpack.ksp.visitor.ViewStoriesVisitor
+import net.skyscanner.backpack.ksp.writer.writeListOfStories
+
+@OptIn(ExperimentalProcessingApi::class)
+class BackpackSymbolProcessor(
```

Comment:
can you add some tests for these? as the ksp files will end up in the build folder we won't have a manual check to see if anything changed, but need to rely on tests to ensure everything is in the right format. particularly with different edge cases, view/compose versions etc